crates/core_arch/src/arm/neon.rs

```diff
@@ -1136,12 +1388,526 @@ pub unsafe fn vtbx4_p8(a: poly8x8_t, b: poly8x8x4_t, c: uint8x8_t) -> poly8x8_t
+#[cfg_attr(all(test, target_arch = "aarch64"), assert_instr(dup))]
+pub unsafe fn vmovq_n_u8(value: u8) -> uint8x16_t {
+    vdupq_n_u8(value)
+}
+
+macro_rules! arm_reinterpret {
+    ($name:ident, $from:ty, $to:ty) => {
+        // Vector reinterpret cast operation
+        #[inline]
+        #[target_feature(enable = "neon")]
+        #[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
+        pub unsafe fn $name(a: $from) -> $to {
+            transmute(a)
+        }
+    };
+}
+
+// uint64x1_t vreinterpret_u64_u32 (uint32x2_t a)
+arm_reinterpret!(vreinterpret_u64_u32, uint32x2_t, uint64x1_t);
+
+// int8x16_t vreinterpretq_s8_u8 (uint8x16_t a)
+arm_reinterpret!(vreinterpretq_s8_u8, uint8x16_t, int8x16_t);
+
+// uint16x8_t vreinterpretq_u16_u8 (uint8x16_t a)
+arm_reinterpret!(vreinterpretq_u16_u8, uint8x16_t, uint16x8_t);
+
+// uint32x4_t vreinterpretq_u32_u8 (uint8x16_t a)
+arm_reinterpret!(vreinterpretq_u32_u8, uint8x16_t, uint32x4_t);
+
+// uint64x2_t vreinterpretq_u64_u8 (uint8x16_t a)
+arm_reinterpret!(vreinterpretq_u64_u8, uint8x16_t, uint64x2_t);
+
+// uint8x16_t vreinterpretq_u8_s8 (int8x16_t a)
+arm_reinterpret!(vreinterpretq_u8_s8, int8x16_t, uint8x16_t);
+
+//uint8x16_t vshrq_n_u8 (uint8x16_t a, const int n)
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
+#[cfg_attr(test, assert_instr(ushr, imm3 = 1))]
+#[rustc_args_required_const(1)]
+pub unsafe fn vshrq_n_u8(a: uint8x16_t, imm3: i32) -> uint8x16_t {
+    if imm3 < 0 || imm3 > 7 {
+        unreachable_unchecked();
+    } else {
+        uint8x16_t(
+            a.0 >> imm3,
+            a.1 >> imm3,
+            a.2 >> imm3,
+            a.3 >> imm3,
+            a.4 >> imm3,
+            a.5 >> imm3,
+            a.6 >> imm3,
+            a.7 >> imm3,
+            a.8 >> imm3,
+            a.9 >> imm3,
+            a.10 >> imm3,
+            a.11 >> imm3,
+            a.12 >> imm3,
+            a.13 >> imm3,
+            a.14 >> imm3,
+            a.15 >> imm3,
+        )
+    }
+}
+
+//uint8x16_t vshlq_n_u8 (uint8x16_t a, const int n)
+#[inline]
+#[target_feature(enable = "neon")]
+#[cfg_attr(target_arch = "arm", target_feature(enable = "v7"))]
+#[cfg_attr(test, assert_instr(shl, imm3 = 1))]
+#[rustc_args_required_const(1)]
+pub unsafe fn vshlq_n_u8(a: uint8x16_t, imm3: i32) -> uint8x16_t {
+    if imm3 < 0 || imm3 > 7 {
+        unreachable_unchecked();
+    } else {
+        uint8x16_t(
+            a.0 << imm3,
+            a.1 << imm3,
+            a.2 << imm3,
+            a.3 << imm3,
+            a.4 << imm3,
+            a.5 << imm3,
+            a.6 << imm3,
+            a.7 << imm3,
+            a.8 << imm3,
+            a.9 << imm3,
+            a.10 << imm3,
+            a.11 << imm3,
+            a.12 << imm3,
+            a.13 << imm3,
+            a.14 << imm3,
+            a.15 << imm3,
+        )
+    }
+}
+
+/// Extract vector from pair of vectors
```

Comment:
Does this comment apply to the function below in the generated docs ?

Reply:
Yes that's the comment from the ARM manual for `vextq_s8`

Reply:
I removed the comment blow to make it more clear